uclapi.openapi.json

```diff
@@ -0,0 +1,4256 @@
+            "type" : "string",
+            "example" : "291",
+            "description": "The ID of the room."
+          },
+          "location" : {
+            "type" : "string",
+            "example" : "",
+            "description": "The location of this sensor."
+          },
+          "survey_device_id" : {
+            "type" : "string",
+            "example" : "8420",
+            "description": "The ID of this survey device."
+          },
+          "share_id" : {
+            "type" : "string",
+            "example" : "None",
+            "description": "The Share ID."
+          },
+          "x_pos" : {
+            "type" : "string",
+            "example" : "32432.0",
+            "description": "The X coordinate position of the sensor on the map."
+          },
+          "description_1" : {
+            "type" : "string",
+            "example" : "Description 1."
+          },
+          "hardware_id" : {
+            "type" : "string",
+            "example" : "584001",
+            "description": "A unique ID for the sensor."
+          },
+          "pir_address" : {
+            "type" : "string",
+            "example" : "1",
+            "description": "PIR address."
+          },
+          "last_trigger_timestamp" : {
+            "type" : "string",
+            "example" : "2018-02-15T22:42:28+00:00",
+            "description": "The last time that this sensor's update value was recorded. NB: only returned when return_states=true."
+          },
+          "occupied" : {
+            "type" : "boolean",
+            "example" : true,
+            "description": "Whether the seat is occupied or not. This takes into account UCL's thirty minute rule, so it's possible that the seat could have been marked as absent a few minutes ago, but this value is still set to true, as UCL allows students to leave their seat unattended for up to thirty minutes at a time (e.g. to use the bathroom or get food). Integrations that provide live seating information to students should use this value so as to have parity with the UCL Library seating policy and the UCL Library website. NB: only returned when return_states=true."
+          }
+        }
+      },
+      "sensor_average_survey" : {
+        "properties" : {
+          "survey_id" : {
+            "type" : "number",
+            "example" : 46,
+            "description": "The ID of the library/survey that was requested in the query parameter."
+          },
+          "name" : {
+            "type" : "string",
+            "example" : "UCL Institute of Education Library",
+            "description": "Name of survey location."
+          },
+          "averages" : {
+            "type" : "object",
+            "additionalProperties" : {
+              "$ref" : "#/components/schemas/average"
+            }
+          }
+        }
+      },
+      "historical_survey_data" : {
+        "properties" : {
+          "sensor_id" : {
+            "type" : "integer",
```

Comment:
nit: integer instead of number (same with `averages`)

Reply:
I think it might be better to be more specific and stick with `integer` here?

Reply:
Looking through the code we have a lot of fields with types that I don't really get, e.g., `weeknumber` is a `FloatField` for `/roombookings`, and `BigIntegerField` for `/timetable`.

we could probably update a lot of our types -- what do you think? if we did #3060 this could go along with it.